config/manager/deployment.yaml

```diff
@@ -41,6 +41,7 @@ spec:
           - --log-json
           - --enable-leader-election
           - --storage-path=/data
+          - --storage-adv-addr=source-controller.$(RUNTIME_NAMESPACE).svc.cluster.local.
```

Comment:
The dot in the end causing issue:
```
failed to download artifact, error: GET http://source-controller.flux-system.svc.cluster.local./gitrepository/flux-system/***/******.tar.gz giving up after 10 attempt(s)

```

Reply:
This should generally not result in any issues, as this is the most absolute FQDN which exists: http://www.dns-sd.org/trailingdotsindomainnames.html

Are you certain your cluster domain is `cluster.local`?